File: packages/sdk-php/src/BugsAPI.php
<?php

declare(strict_types=1);

namespace Melaya;

/**
 * Bugs API — submit and track bug reports.
 *
 * Maps to /api/v1/private/bugs/*.
 *
 * @example
 * ```php
 * $sdk = new Melaya\Melaya(apiKey: getenv('MK'));
 *
 * $bug = $sdk->bugs->create(['title' => 'Pipeline stuck', 'description' => '...']);
 * $sdk->bugs->addComment($bug['id'], 'Still reproducible on v2.3');
 * ```
 */
class BugsAPI
{
    public function __construct(private readonly HttpClient $http) {}

    /**
     * Submit a bug report (user-facing feedback form).
     *
     * @param array $body ['title' => '...', 'description' => '...', 'severity' => '...']
     */
    public function create(array $body): array
    {
        return $this->http->post('/api/v1/private/bugs', $body);
    }

    /** List bug reports submitted by the caller. */
    public function listMine(): array
    {
        return $this->http->get('/api/v1/private/bugs/mine');
    }

    /** Get a single bug report by ID. */
    public function get(string $bugId): array
    {
        return $this->http->get('/api/v1/private/bugs/' . rawurlencode($bugId));
    }

    /**
     * Add a comment to a bug report.
     *
     * @param string $bugId   Bug report ID
     * @param string $comment Comment text
     */
    public function addComment(string $bugId, string $comment): array
    {
        return $this->http->post(
            '/api/v1/private/bugs/' . rawurlencode($bugId) . '/comments',
            ['comment' => $comment]
        );
    }

    /** List unread bug-related notifications for the caller. */
    public function listNotifications(): array
    {
        return $this->http->get('/api/v1/private/bugs/notifications');
    }

    /** Mark bug notifications as read. */
    public function markNotificationsRead(array $body = []): array
    {
        return $this->http->post('/api/v1/private/bugs/notifications/read', $body ?: null);
    }
}

File: packages/sdk-csharp/Melaya/BugsApi.cs
using System.Text.Json;

namespace Melaya;

/// <summary>
/// Bugs API — submit bug reports and read notifications.
/// Maps to <c>/api/v1/private/bugs/*</c>.
/// </summary>
public sealed class BugsApi
{
    private readonly MelayaHttpClient _http;

    internal BugsApi(MelayaHttpClient http) => _http = http;

    /// <summary>Submit a bug report.</summary>
    public async Task<BugReport> CreateAsync(BugReportCreateRequest body, CancellationToken ct = default)
    {
        return await _http.PostAsync<BugReport>("/api/v1/private/bugs", body, ct).ConfigureAwait(false);
    }

    /// <summary>List bug reports submitted by the caller.</summary>
    public async Task<List<BugReport>> ListMineAsync(CancellationToken ct = default)
    {
        return await _http.GetAsync<List<BugReport>>("/api/v1/private/bugs/mine", ct: ct).ConfigureAwait(false);
    }

    /// <summary>Get a single bug report by ID.</summary>
    public async Task<BugReport> GetAsync(string bugId, CancellationToken ct = default)
    {
        return await _http.GetAsync<BugReport>(
            $"/api/v1/private/bugs/{Uri.EscapeDataString(bugId)}", ct: ct).ConfigureAwait(false);
    }

    /// <summary>Add a comment to a bug report.</summary>
    public async Task<JsonElement> AddCommentAsync(string bugId, string comment, CancellationToken ct = default)
    {
        var body = new { comment };
        return await _http.PostAsync<JsonElement>(
            $"/api/v1/private/bugs/{Uri.EscapeDataString(bugId)}/comments", body, ct).ConfigureAwait(false);
    }

    /// <summary>List unread bug-related notifications for the caller.</summary>
    public async Task<List<JsonElement>> ListNotificationsAsync(CancellationToken ct = default)
    {
        return await _http.GetAsync<List<JsonElement>>("/api/v1/private/bugs/notifications", ct: ct).ConfigureAwait(false);
    }

    /// <summary>Mark bug notifications as read.</summary>
    public async Task<BoolResult> MarkNotificationsReadAsync(CancellationToken ct = default)
    {
        return await _http.PostAsync<BoolResult>("/api/v1/private/bugs/notifications/read", null, ct).ConfigureAwait(false);
    }
}
